src/providers/aws.rs

```diff
@@ -241,6 +241,7 @@ pub struct Launcher<P = DefaultCredentialsProvider> {
     max_instance_duration_hours: usize,
     use_open_ports: bool,
     regions: HashMap<<Setup as super::MachineSetup>::Region, RegionLauncher>,
+    rt: Option<tokio::runtime::Runtime>,
```

Comment:
Any particular reason to make this an `Option`?

Reply:
Because of the `impl Drop`, it can't be destructured, so the Option makes https://github.com/jonhoo/tsunami/pull/22/files/df53969d0e11c5b97abc9b6cd06dd9a4e8e441da..ccb5f748dea088cddafb6a4b02d24a5cef73e933#diff-60699702750aef5a8cdc050b4fce64eaR297 work. Is there a better way?

Reply:
Ah, hmm, that's awkward. In its current form, that won't work anyway, since if you call `with_credentials`, the runtime is taken from the old `self`, which is then dropped, and its `Drop` call will then fail the `expect` on `self.rt`.

Come to think of it, why aren't the various configuration options on `Setup` instead of `Launcher`? Seems to me like once you have a `Launcher` all the settings should already have been set up, including credentials? That'd solve our problem I think.

Reply:
I think it is safe (added a code comment with my reasoning)? Also, it seems silly to per-machine define something which naturally applies to the whole launcher.

Reply:
Ah, sorry, I was thinking about a launcher builder, but that's not what `Setup` is... Hmm.. I guess it's probably fine.